### ISSUE TITLE
Prevent XML parser from parsing gzipped XMLs that it's unable to decompress

### ISSUE THREAD

**Issue by pypt** [berkmancenter/mediacloud-ultimate_sitemap_parser#6]
```
2018-11-26 12:59:27,847 INFO mediawords.util.sitemap.fetchers
[194712/MainThread]: Fetching level 1 sitemap from
https://www.iberlibro.com/sitemap.bdp31.xml.gz...
2018-11-26 12:59:27,848 INFO mediawords.util.sitemap.helpers
[194712/MainThread]: Fetching URL
https://www.iberlibro.com/sitemap.bdp31.xml.gz...
2018-11-26 12:59:28,433 ERROR mediawords.util.sitemap.helpers
[194712/MainThread]: Unable to gunzip response
<mediawords.util.web.user_agent.response.response.Response object at
0x7f3485abfcc8>: Unable to gunzip data: Not a gzipped file (b'<?')
2018-11-26 12:59:28,437 INFO mediawords.util.sitemap.fetchers
[194712/MainThread]: Parsing sitemap from URL
https://www.iberlibro.com/sitemap.bdp31.xml.gz...
```

**Comment by probar** [comment]
This issue causes some partial fetching of some sitemaps.

Shouldn't that be an open bug ?

**Comment by pypt** [comment]
It should have been fixed in 3c2b076.

Please reopen / create a new issue if you still encounter this. Examples with live websites would be helpful too!

**Comment by probar** [comment]
Sorry for the late reply. 

The way you fixed this in 3c2b076 , is correctly, rejecting bad gz files. 

According to [1], that issue might happen because a misconfig of the server, and one optional way to graciously handle this issue, is to download the uncompressed xml file. 

Does it sound reasonable  ? 

[1]https://www.microsystools.com/products/sitemap-generator/help/xml-sitemaps-compression-gzip/
